tests/genomics/antismash/test_downloader.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+from nplinker.genomics.antismash import download_and_extract_antismash_metadata
+
+
+class TestDownloadAndExtractAntismashData():
+
+    refseq_assembly_id = "GCF_004339725.1"
+
+    def test_default(self, tmp_path):
+        download_root = tmp_path / "download"
+        download_root.mkdir()
+        extract_root = tmp_path / "extracted"
+        extract_root.mkdir()
+        download_and_extract_antismash_metadata(self.refseq_assembly_id,
+                                                download_root, extract_root)
+        archive = download_root / "GCF_004339725.1.zip"
+        extracted_folder = extract_root / "antismash/GCF_004339725.1"
+        assert archive.exists()
+        assert archive.is_file()
+        assert extracted_folder.exists()
+
+    def test_error_same_path(self, tmp_path):
+        with pytest.raises(ValueError) as e:
+            download_and_extract_antismash_metadata(self.refseq_assembly_id,
+                                                    tmp_path, tmp_path)
+        assert e.value.args[
+            0] == "Identical path of download directory and extract directory"
+
+    def test_error_nonempty_path(self, tmp_path):
+        nonempty_path = tmp_path / f"extracted/antismash/{self.refseq_assembly_id}/subdir"
```

Comment:
```suggestion
        nonempty_path = tmp_path / "extracted" / "antismash" / f"{refseq_assembly_id}" / "subdir"
```

Do not use fixed separator between path, instead let `Pathlib.Path` object (like `tmp_path` here) to decide it automatically.